01_stratified_classification/nbconverted/lasso_range_gene_optimizers.py

```diff
@@ -0,0 +1,791 @@
+    .drop(columns=['holdout_cancer_type', 'experiment'])
+    .copy()
+)
+ll_perf_df.head()
+
+
+# In[7]:
+
+
+sgd_perf_df = au.load_prediction_results_lasso_range(sgd_results_dir,
+                                                     'stratified',
+                                                     gene=plot_gene)
+sgd_perf_df = (
+    sgd_perf_df[sgd_perf_df.gene == plot_gene]
+    .drop(columns=['holdout_cancer_type', 'experiment'])
+    .copy()
+)
+sgd_perf_df.head()
+
+
+# In[8]:
+
+
+sns.set_style('ticks')
+
+ll_plot_df = (
+    ll_perf_df[(ll_perf_df.signal == 'signal')]
+      .sort_values(by=['lasso_param'])
+      .reset_index(drop=True)
+)
+ll_plot_df.lasso_param = ll_plot_df.lasso_param.astype(float)
+
+sgd_plot_df = (
+    sgd_perf_df[(sgd_perf_df.signal == 'signal')]
+      .sort_values(by=['lasso_param'])
+      .reset_index(drop=True)
+)
+sgd_plot_df.lasso_param = sgd_plot_df.lasso_param.astype(float)
+
+ll_plot_df['optimizer'] = 'liblinear'
+sgd_plot_df['optimizer'] = 'SGD'
+
+plot_df = pd.concat((ll_plot_df, sgd_plot_df))
+
+with sns.plotting_context('notebook', font_scale=1.6):
+    g = sns.relplot(
+        data=plot_df,
+        x='lasso_param', y=metric, hue='data_type',
+        hue_order=['train', 'cv', 'test'],
+        marker='o', kind='line', col='optimizer',
+        col_wrap=2, height=5, aspect=1.6,
+        facet_kws={'sharex': False}
+    )
+    g.set(xscale='log', xlim=(min(plot_df.lasso_param), max(plot_df.lasso_param)))
+    g.axes[0].set_xlabel('LASSO parameter (higher = less regularization)')
+    g.axes[0].set_xlim((10e-4, 10e2))
+    g.axes[1].set_xlabel('LASSO parameter (lower = less regularization)')
+    g.axes[1].set_xlim((10e-7, 10))
+    g.set_ylabels(f'{metric.upper()}')
+    sns.move_legend(g, "center", bbox_to_anchor=[1.035, 0.55], frameon=True)
+    g._legend.set_title('Dataset')
+    new_labels = ['train', 'holdout', 'test']
+    for t, l in zip(g._legend.texts, new_labels):
+        t.set_text(l)
+    g.set_titles('Optimizer: {col_name}')
+     
+    plt.suptitle(f'LASSO parameter vs. {metric.upper()}, {plot_gene}', y=1.0)
+
+plt.tight_layout()
+
+if output_plots:
+    plt.savefig(os.path.join(output_plots_dir, f'{gene}_parameter_vs_perf.svg'), bbox_inches='tight')
+
+
+# In[9]:
+
+
+ll_nz_coefs_df['optimizer'] = 'liblinear'
+sgd_nz_coefs_df['optimizer'] = 'SGD'
+
+nz_coefs_df = pd.concat((ll_nz_coefs_df, sgd_nz_coefs_df))
+nz_coefs_df.head()
+
+
+# In[10]:
+
+
+perf_coefs_df = (plot_df
+    .merge(nz_coefs_df,
+           left_on=['gene', 'optimizer', 'lasso_param', 'seed', 'fold'],
+           right_on=['gene', 'optimizer', 'lasso_param', 'seed', 'fold'])
+)
+
+print(perf_coefs_df.shape)
+perf_coefs_df.head()
+
+
+# In[11]:
+
+
+sns.set({'figure.figsize': (10, 5)})
+sns.set_style('ticks')
+sns.histplot(perf_coefs_df.nz_coefs)
+plt.title(f'Nonzero coefficient distribution for {plot_gene}')
+plt.xlabel('Number of nonzero coefficients')
+
+linear_bins_df = []
```

Comment:
This list isn't getting used anywhere, and I'm not sure what the linear bins are anyway. They don't seem to match up with the bins being auto-generated for the histogram?

Reply:
I guess it makes sense to have them be the same, I was just using the default number of bins